src/pipeline.js

```diff
@@ -0,0 +1,35 @@
+//
+// based on:
+//  https://blakeembrey.com/articles/2014/01/compose-functions-javascript/
+//
+// but done in the opposite order, so it looks like an elixir pipeline: 
+//  "a |> b |> c"
+//
+// so:
+//  pipeline(a, b, c)()
+// would be the same as:
+//  c(b(a()))
+//
+// rather than looping through an array of functions:
+//
+//   for (var i = 0; i < functions.length; i++) {
+//      result = functions[i].call(this, result);
+//   }
+//
+// we're using the Array.reduce method, which calls a callback
+// each time, passing an accumulated result back into the callback
+//
+// the second argument to the callback is the next function in the array
+//
+
+function pipeline () {
```

Comment:
@cookiescrumbs I've gone down a right rabbit hole with this one, so using "reduce" to loop through an array of functions, I think it might just be easier to read if `.then(flatten).then(transform).then(model)` than having to understand what this is doing. Trying to be clever with higher-order functions 👉 🍆